RocketChat/Rocket.Chat.ReactNative
# fix: the download's key for files with the same filename

## Proposed changes


The error occurred when there were more than 1 file with the same filename, for example:

`https://open.rocket.chat/file-upload/ChJDEKZhJHc359rm8/Screenshot%202024-02-14%20at%2019.54.07.png`

`https://open.rocket.chat/file-upload/4roCoCiHzsxzZzFXv/Screenshot%202024-02-14%20at%2019.54.07.png`

The `mediaDownloadKey` function is responsible for determining the key related to the download object, but it was only capturing the filename from the URL. Thus, when there were 2 files with the same filename, the key was associated with only one, causing the download of the other to appear as if it was also in progress, when in fact, it was referencing the download of the first file. To fix this, I decided not to separate only the filename to use as the key for the object. Instead, I used the complete link as the object key, thus differentiating between each file.

## Issue(s)	


## How to test or reproduce


- Upload two different files with the same name
- When them appears in the screen, you need to click one by one almost at the same time
- Then after the download is done and you click to open the file, you will notice that you are opening the same file
- One point to consider is: if you exit the channel and reopen it, you will notice that one of the files wasn't downloaded. The download icon will appear, and after clicking to download it, the file should be downloaded as expected.

## Screenshots

#### Before

https://github.com/RocketChat/Rocket.Chat.ReactNative/assets/47038980/97c666cb-ca4e-4a91-9991-f513322a1148

#### After

https://github.com/RocketChat/Rocket.Chat.ReactNative/assets/47038980/5b773be2-fb99-465e-a971-9df9baf75d84


## Types of changes




## Checklist



## Further comments

[CORE-55]

[CORE-55]: https://rocketchat.atlassian.net/browse/CORE-55?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
